google-ads/src/main/java/com/google/ads/googleads/lib/GoogleAdsClient.java

```diff
@@ -552,9 +552,7 @@ public GoogleAdsClient build() {
       Primer.getInstance().ifPresent(p -> p.primeCredentialsAsync(getCredentials()));
       // Proceeds with creating the client library instance.
       TransportChannelProvider transportChannelProvider = getTransportChannelProvider();
-      if (transportChannelProvider.needsHeaders()) {
-        transportChannelProvider = transportChannelProvider.withHeaders(getHeaders());
-      }
+      transportChannelProvider = transportChannelProvider.withHeaders(getHeaders());
```

Comment:
You could chain this with line 554. Up to you if think that helps or hinders readability.

Reply:
Good idea, updated.